internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -301,25 +295,28 @@ func resourceRecoveryServicesVaultUpdate(d *pluginsdk.ResourceData, meta interfa
 	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id := parse.NewVaultID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
+	id := vaults.NewVaultID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
 
 	encryption := expandEncryption(d)
-	existing, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	existing, err := client.Get(ctx, id)
 	if err != nil {
 		return fmt.Errorf("checking for presence of existing Recovery Service %s: %+v", id.String(), err)
 	}
-	if existing.Properties != nil && existing.Properties.Encryption != nil {
-		if encryption == nil {
-			return fmt.Errorf("once encryption with your own key has been enabled it's not possible to disable it")
-		}
-		if encryption.InfrastructureEncryption != existing.Properties.Encryption.InfrastructureEncryption {
-			return fmt.Errorf("once `infrastructure_encryption_enabled` has been set it's not possible to change it")
-		}
-		if d.HasChange("sku") {
-			// Once encryption has been enabled, calling `CreateOrUpdate` without it is not allowed.
-			// But `sku` can only be updated by `CreateOrUpdate` and the support for `encryption` in `CreateOrUpdate` is still under preview (https://docs.microsoft.com/azure/backup/encryption-at-rest-with-cmk?tabs=portal#enable-encryption-using-customer-managed-keys-at-vault-creation-in-preview).
-			// TODO remove this restriction and add `encryption` to below `sku` update block when `encryption` in `CreateOrUpdate` is GA
-			return fmt.Errorf("`sku` cannot be changed when encryption with your own key has been enabled")
+	if existing.Model != nil && existing.Model.Properties != nil {
+		prop := existing.Model.Properties
+		if prop.Encryption != nil {
+			if encryption == nil {
+				return fmt.Errorf("once encryption with your own key has been enabled it's not possible to disable it")
+			}
+			if *encryption.InfrastructureEncryption != *prop.Encryption.InfrastructureEncryption {
+				return fmt.Errorf("once `infrastructure_encryption_enabled` has been set it's not possible to change it")
+			}
+			if d.HasChange("sku") {
+				// Once encryption has been enabled, calling `CreateOrUpdate` without it is not allowed.
+				// But `sku` can only be updated by `CreateOrUpdate` and the support for `encryption` in `CreateOrUpdate` is still under preview (https://docs.microsoft.com/azure/backup/encryption-at-rest-with-cmk?tabs=portal#enable-encryption-using-customer-managed-keys-at-vault-creation-in-preview).
+				// TODO remove this restriction and add `encryption` to below `sku` update block when `encryption` in `CreateOrUpdate` is GA
```

Comment:
is this restriction still required?

Reply:
Take a look at the feature and it's still in preview. And this PR is only to swap SDK without upgration of API version. Let me veritfy it in detail in another PR to upgrade API version. Does it sound fair?